src/cysignals/implementation_cygwin.h

```diff
@@ -0,0 +1,61 @@
+/* Cygwin-specific implementation details */
```

Comment:
This should be a `.c` file instead of `.h`

Reply:
That was my original call as well--it was a ".c" file to start with and I had `#include "implementation_cygwin.c"`.  However, I found out the hard way that in this case re-running `setup.py` didn't rebuild `implementation.o`, which is annoying.  Open to ideas.

Reply:
> I found out the hard way that in this case re-running setup.py didn't rebuild implementation.o, which is annoying.

That issue is independent of the filename.

Reply:
You would think, but when I changed the filename that seemed to fix it.  Maybe it was something else.  
I think that, rather than doing an "include" I'd rather just make this a separate (optional) compilation unit.  However, that will require adding a couple header files.